cirq-core/cirq/sim/classical_simulator.py

```diff
@@ -12,25 +12,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict
-from collections import defaultdict
-from cirq.sim.simulator import SimulatesSamples
-from cirq import ops, protocols
-from cirq.study.resolver import ParamResolver
-from cirq.circuits.circuit import AbstractCircuit
+from cirq import ops
 from cirq.ops.raw_types import Qid
-import numpy as np
+from typing import List, Sequence
+from cirq.sim.simulation_state import SimulationState
+from cirq import qis
+from cirq.value import big_endian_int_to_bits
 
 
-def _is_identity(op: ops.Operation) -> bool:
-    if isinstance(op.gate, (ops.XPowGate, ops.CXPowGate, ops.CCXPowGate, ops.SwapPowGate)):
-        return op.gate.exponent % 2 == 0
+def _is_identity(gate: ops.GateOperation) -> bool:
+    if isinstance(gate, (ops.XPowGate, ops.CXPowGate, ops.CCXPowGate, ops.SwapPowGate)):
+        return gate.exponent % 2 == 0
     return False
 
 
-class ClassicalStateSimulator(SimulatesSamples):
-    """A simulator that accepts only gates with classical counterparts.
+class ClassicalState(qis.QuantumStateRepresentation):
+    def __init__(self, initial_state: List[int]):
+        self.basis = initial_state
+
+    def copy(self, deep_copy_buffers: bool = True) -> 'ClassicalState':
+        return ClassicalState(self.basis)
+
+    def measure(self, axes: Sequence[int], seed: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None):
+        return [self.basis[i] for i in axes]
 
+
+class ClassicalStateSimulator(SimulationState[ClassicalState]):
```

Comment:
The parent class is wrong. The `ClassicalStateSimulator` is a simulator not a state. The class should resemble `CustomStateSimulator` from https://github.com/daxfohl/Cirq/blob/bf01bb447665acf49c34f5652626552c2b5a711a/cirq-core/cirq/contrib/custom_simulators/custom_state_simulator.py#L40